recipes/gsoap/all/conanfile.py

```diff
@@ -16,28 +18,56 @@ class GsoapConan(ConanFile):
     topics = ("gsoap", "logging")
     settings = "os", "arch", "compiler", "build_type"
     options = {
+        "fPIC": [True, False],
         "with_openssl": [True, False],
         "with_ipv6": [True, False],
         "with_cookies": [True, False],
         "with_c_locale": [True, False],
     }
     default_options = {
-        'with_openssl': True,
-        'with_ipv6': True,
-        'with_cookies': True,
-        'with_c_locale': True,
+        "fPIC": True,
+        "with_openssl": True,
+        "with_ipv6": True,
+        "with_cookies": True,
+        "with_c_locale": True,
     }
+
+    exports_sources = "CMakeLists.txt", "cmake/*.cmake"
     short_paths = True
 
-    def export_sources(self):
-        tools.files.copy(self, "CMakeLists.txt", self.recipe_folder, self.export_sources_folder)
-        tools.files.copy(self, "*.cmake", os.path.join(self.recipe_folder, "src"), os.path.join(self.export_sources_folder, "src"))
-        for p in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.files.copy(self, p["patch_file"], self.recipe_folder, self.export_sources_folder)
+    @property
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.options.with_openssl:
+            self.requires("openssl/1.1.1q")
+            self.requires("zlib/1.2.12")
+
+    def build_requirements(self):
+        if cross_building(self, skip_x64_x86=True) and hasattr(self, "settings_build"):
+            self.tool_requires("gsoap/{}".format(self.version))
+
+        if self._settings_build.os == "Windows":
+            self.tool_requires("winflexbison/2.5.24")
+        else:
+            self.tool_requires("bison/3.7.6")
+            self.tool_requires("flex/2.6.4")
```

Comment:
It's worth noting that this logic is broken for MinGW@windows, since winflexbison can only be built with Visual Studio.
We should fix flex recipe to be able to build with MinGW (it's already the case in bison recipe), and replace build requirements of recipes relying on flex & bison with:

```
        if is_msvc(self):
            self.tool_requires("winflexbison/2.5.24")
        else:
            self.tool_requires("bison/3.7.6")
            self.tool_requires("flex/2.6.4")
```

Reply:
you are right. At the time I wrote the recipe I couldn't get tested on MinGW and for personal usage I could only test the static library build. Thanks for your effort here